netplan_cli/cli/commands/sriov_rebind.py

```diff
@@ -44,7 +66,105 @@ def command_rebind(self):
+                        # If the PF is a member of a bond, the user might be trying to enable the
+                        # VF LAG feature.
+                        # Mellanox VF LAG requires that the LAG state reports as 'active'
+                        # *before* VFs can be bound to the driver. Performing the bind operation
+                        # before the device is ready will cause the VF LAG feature to never be enabled.
+
+                        # Another condition for the VF LAG activation is that the LAG mode
+                        # must be one of 'active-backup', 'balanced-xor' or '802.3ad'.
+                        bond_mode = bond_link._bond_mode
+                        if not self._is_bond_mode_supported(bond_mode):
+                            logging.debug(f'{iface} - LAG mode {bond_mode} is not supported by VF LAG')
+                            continue
+
+                        logging.debug(f'{iface} - waiting for the LAG state to be \'active\'')
+                        try:
+                            self._wait_for_mlx5_pf_lag_state_active(pf)
+                        except MLX5VFLAGStateCannotBeRead:
+                            logging.debug(f'{iface} - VF LAG state cannot be read')
+                        except MLX5VFLAGStateNotFound:
+                            logging.debug(f'{iface} - VF LAG state debugfs file not found')
+                        except MLX5VFLAGStateDisabled:
+                            logging.debug(f'{iface} - VF LAG state is still \'disabled\' after waiting')
+                        else:
+                            logging.debug(f'{iface} - VF LAG state is \'active\'')
+
+    def _wait_for_mlx5_pf_lag_state_active(self, pf: PCIDevice):
+        """
+        The mlx5 driver added support for debugfs in https://github.com/torvalds/linux/commit/7f46a0b7327a
+        It's available since kernel 5.19 https://cdn.kernel.org/pub/linux/kernel/v5.x/ChangeLog-5.19
+        """
+        retries = int(MAX_WAITING_TIME_SEC / INTERVAL_SEC)
+        pci_addr = pf.pci_addr
+        path = f'/sys/kernel/debug/mlx5/{pci_addr}/lag/state'
+
+        if not os.path.exists(path):
+            # If the debugfs file doesn't exist, it might be because this version of the mlx5 driver
+            # still doesn't support it or because the debugfs is not mounted.
+            # In this case, we probably should still wait for a few seconds to give time for the
+            # driver to change state.
+            # Based on tests with a ConnectX-5 NIC, 1 second is enough time, so let's wait a bit more
+            # just in case. This delay will only be introduced if the PF is part of a bond.
+            sleep(FALLBACK_WAIT_TIME_SEC)
+            raise MLX5VFLAGStateNotFound
+
+        while retries > 0:
+            try:
+                if self._get_mlx5_vf_lag_state(pci_addr) != 'active':
+                    logging.debug(f'{pci_addr} VF LAG state is not active yet, retrying in 1 second...')
```

Comment:
nitpick (non-blocking): We're retrying in `INTERVAL_SEC` seconds. Just drop the specific time from the log message... "VF LAG state is not active yet, retrying..."